sdk/android/src/jni/pc/frame_cryptor_key_manager.h

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2022 LiveKit
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SDK_ANDROID_SRC_JNI_PC_FRAME_CRYPTOR_KEY_MANAGER_H_
+#define SDK_ANDROID_SRC_JNI_PC_FRAME_CRYPTOR_KEY_MANAGER_H_
+
+#include <jni.h>
+
+#include "api/crypto/frame_crypto_transformer.h"
+#include "sdk/android/native_api/jni/scoped_java_ref.h"
+
+namespace webrtc {
+namespace jni {
+
+class DefaultKeyManagerImpl : public webrtc::KeyManager {
+ public:
+  DefaultKeyManagerImpl() = default;
+  ~DefaultKeyManagerImpl() override = default;
+
+  /// Set the key at the given index.
+  bool SetKey(const std::string participant_id,
+              int index,
+              std::vector<uint8_t> key) {
+    if (index > webrtc::KeyManager::kMaxKeySize) {
+      return false;
+    }
+
+    if (keys_.find(participant_id) == keys_.end()) {
+      keys_[participant_id] = std::vector<std::vector<uint8_t>>();
+    }
+
+    webrtc::MutexLock lock(&mutex_);
```

Comment:
Shouldn't the lock be at the start of the function?

Reply:
Yes, webrtc::FrameEncryptorInterface is missing the key information of FrameCrypto, codec type (VP8, H264 use different encryption methods), frame type, ssrc and timestamp (for building IV)

Reply:
You are right, you should lock at the beginning of the function line